activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -327,6 +327,80 @@ def reset_sequence!(table, column, sequence = nil)
         # Do nothing by default. Implement for PostgreSQL, Oracle, ...
       end
 
+      def insert_many(inserts, table_name, options = {})
```

Comment:
This method is 37 lines long. I think there's support for breaking it into a command object, so we can handle all the cases without a mega method.